src/backend/distributed/utils/node_metadata.c

```diff
@@ -106,7 +107,16 @@ master_remove_node(PG_FUNCTION_ARGS)
 	EnsureSuperUser();
 
 	hasShardPlacements = NodeHasActiveShardPlacements(nodeNameString, nodePort);
-	if (hasShardPlacements)
+	if (hasShardPlacements && force)
+	{
+		ereport(NOTICE, (errmsg("Node %s:%d has active shard placements. Some "
+								"queries may fail after this operation. Use "
+								"select master_add_node('%s', %d) to add this "
+								"node back.",
+								nodeNameString, nodePort, nodeNameString,
+								nodePort)));
+	}
+	else if (hasShardPlacements)
 	{
 		ereport(ERROR, (errmsg("you cannot remove a node which has active "
```

Comment:
Rewrite this message to explain that `force` is necessary if you wish to do this.